### ISSUE TITLE
Using Fulfill to send QuickReply buttons in combination with <pause/> tag on FB Messenger - timing is wrong

### ISSUE THREAD

**Issue by duncsand** [botmasterai/botmaster-fulfill#9]
Ho Ho Ho!\<pause />\<greet tz="Europe/London" />, Santa here what would you like to know? \<pause />\<buttons>Yes,No\</buttons>

In the above example, the timing of the quick reply buttons appearing in Facebook Messenger is between the "Ho Ho Ho!" and the "Santa here what would you like to know?" messages. Obviously, the expectation is that the buttons come out last, but they don't.

The button implementation is as follows:
```buttons: {
        controller: (params) => {
            const buttonTitles = params.content.split(',');
            params.update.message.quick_replies = [];
            for (const buttonTitle of buttonTitles) {
                params.update.message.quick_replies.push({
                    content_type: 'text',
                    title: buttonTitle,
                    payload: buttonTitle,
                });
            }
            return '';
        },
```

**Comment by jwickens** [comment]
Try the following,
```js
controller: (params) => {
            const buttonTitles = params.content.split(',');
            process.nextTick(function() {
                 params.bot.sendDefaultButtonMessageTo( buttonTitles, params.update.sender.id);
             });
            return '';
        },
```

This will use the botmaster method to send the buttons which combined with the process.nextTick  should send the buttons after the tag has been evaluated.

**Comment by jwickens** [comment]
Hi this has now been fixed in botmaster-fulfill 2.3.0 with a new feature that allows you to run functions after fulfill has finished running.

The syntax is the following

```js
        controller: (params, next) => {
            const buttonTitles = params.content.split(',');
            next().then(() => {
                params.bot.sendMessageTo(buttonTitles, params.update.sender.id)
            });
            return '';
        },
```
Please note that there is a problem with quick replies on facebook when not in messenger.
https://community.chatfuel.com/t/quick-reply-not-showing-in-messages-when-user-are-in-facebook/2069
